src/lib/components/Radio/Player.svelte

```diff
@@ -0,0 +1,246 @@
+      // The divisor (e.g., 400) controls the vibration intensity. Higher value = less intense.
+      const scale = 1 + averageVolume / 400;
+      if (grillElement) {
+        grillElement.style.transform = `scale(${scale})`;
+      }
+
+      // --- Canvas Waveform (unchanged) ---
+      const waveformData = new Uint8Array(analyser.fftSize);
+      analyser.getByteTimeDomainData(waveformData);
+      const { width, height } = canvas;
+      ctx.clearRect(0, 0, width, height);
+      ctx.fillStyle = '#000';
+      ctx.fillRect(0, 0, width, height);
+      ctx.save();
+      ctx.font = 'bold 180px Tahoma, sans-serif';
+      ctx.textAlign = 'center';
+      ctx.textBaseline = 'middle';
+      const cx = width / 2;
+      const cy = height / 2;
+      ctx.fillStyle = '#222';
+      ctx.fillText('AMC', cx, cy);
+      ctx.globalCompositeOperation = 'source-in';
+      ctx.lineWidth = 3;
+      ctx.strokeStyle = 'hsl(200, 100%, 70%)';
+      ctx.beginPath();
+      const slice = width / waveformData.length;
+      let x = 0;
+      for (let i = 0; i < waveformData.length; i++) {
+        const v = waveformData[i] / 128;
+        const y = (v * height) / 2;
+        if (i === 0) ctx.moveTo(x, y);
+        else ctx.lineTo(x, y);
+        x += slice;
+      }
+      ctx.stroke();
+      ctx.restore();
+    }
+    draw();
+  });
+
+  onDestroy(() => {
+    cancelAnimationFrame(animationId);
+    audioCtx?.close();
+  });
+
+  function togglePlay() {
+    // Resume AudioContext if it was suspended
+    if (audioCtx.state === 'suspended') {
+      audioCtx.resume();
+    }
+    if (!isPlaying) audio.play().catch(console.error);
+    else audio.pause();
+    isPlaying = !isPlaying;
+    dispatch('playstate', { isPlaying });
+  }
+
+  function onVolume(e: Event) {
+    volume = +(e.target as HTMLInputElement).value;
+    audio.volume = volume;
+  }
+</script>
+
+<div
+  class="radio-bg border-3 mx-auto flex max-h-screen w-full max-w-[700px] flex-col overflow-hidden rounded-[10px] border-[#5a2c00] shadow-[0_10px_20px_rgba(0,0,0,0.3)] md:h-auto"
+>
+  <div
+    class="border-b-2 border-black/20 bg-[#5a2c00] px-4 py-1.5 text-center font-medium text-[#d2b48c] shadow-[inset_0_-2px_5px_rgba(0,0,0,0.2)]"
+  >
+    {stationName}
+  </div>
+
+  <div class="flex flex-1 flex-col bg-[#d2b48c] md:flex-row">
+    <div
+      class="left-half-bg flex h-[40vh] items-center justify-center p-4 md:h-auto md:flex-1 md:border-r-2 md:border-black/10"
+    >
+      <div bind:this={grillElement} class="grill h-[85%] md:h-[85%] md:w-[85%]"></div>
+    </div>
+
+    <div class="flex h-[40vh] flex-col p-4 md:h-auto md:flex-1">
+      <div
+        class="mb-2.5 flex h-full min-h-[150px] flex-[2] items-center justify-center overflow-hidden rounded-md border-2 border-[#5a2c00] bg-black shadow-[inset_0_0_10px_rgba(0,0,0,0.5)]"
+      >
+        <canvas bind:this={canvas} width="400" height="250" class="block h-auto w-full"></canvas>
```

Comment:
If you don't scale this with device pixel ratio it will be blurry on hidpi display (like mac and phone). See fix https://gist.github.com/callumlocke/cc258a193839691f60dd